spec/chore/worker_spec.rb

```diff
@@ -2,6 +2,28 @@
 
 describe Chore::Worker do
 
+  class FakeDalli
+    def initialize
+      @store = {}
+    end
+    def add(id, val, ttl)
+      if @store[id]
+        return false
+      else
+        @store[id] = val
+        Thread.new do
+          sleep ttl
+          @store.delete(id)
+        end
+        return true
+      end
+    end
+  end
+
+  before(:each) do
+    allow_any_instance_of(Chore::DuplicateDetector).to receive(:memcached_client).and_return(FakeDalli.new)
```

Comment:
Rather than stub the internal `memcached_client` method, we should be able to stub `Dalli::Client.new`.  That would give us the same behavior without having to worry about the internals of the duplicate detector.